api/fossa/normalize2.go

```diff
@@ -0,0 +1,103 @@
+package fossa
+
+import (
+	"github.com/fossas/fossa-cli/graph"
+	"github.com/fossas/fossa-cli/module"
+	"github.com/fossas/fossa-cli/pkg"
+)
+
+type ApiModule struct {
+	Filepath module.Filepath `json:"filepath"`
+	Analyses []ApiAnalysis   `json:"analyses"`
+}
+
+type ApiAnalysis struct {
+	AnalyzerName string           `json:"analyzer"`
+	Graphs       []ApiTaggedGraph `json:"strategies"`
+}
+
+type ApiTaggedGraph struct {
+	Strategy   string   `json:"strategy"`
+	TargetFile string   `json:"targetFile"`
+	Deps       []ApiDep `json:"depgraph"`
+}
+
+type ApiDep struct {
+	Type     string `json:"type"`
+	Name     string `json:"name"`
+	Revision string `json:"revision"`
+
+	Location     string   `json:"location,omitempty"`
+	Dependencies []ApiDep `json:"dependencies,omitempty"`
+}
+
+// TODO: remove these
+//Locator string   `json:"locator"`
+//Imports []string `json:"imports,omitempty"`
+
+func ApiFormatModules(scanned map[module.Filepath][]module.Analysis) []ApiModule {
+	var result []ApiModule
+
+	for path, analyses := range scanned {
+		result = append(result, ApiModule{
+			Filepath: path,
+			Analyses: ApiFormatAnalyses(analyses),
+		})
+	}
+
+	return result
+}
+
+func ApiFormatAnalyses(analyses []module.Analysis) []ApiAnalysis {
+	var result []ApiAnalysis
+
+	for _, analysis := range analyses {
+		result = append(result, ApiAnalysis{
+			AnalyzerName: analysis.AnalyzerName,
+			Graphs:       ApiFormatGraphs(analysis.Graphs),
+		})
+	}
+
+	return result
+}
+
+func ApiFormatGraphs(graphs []module.TaggedGraph) []ApiTaggedGraph {
+	var result []ApiTaggedGraph
+
+	for _, graph := range graphs {
+		result = append(result, ApiTaggedGraph{
+			Strategy:   graph.Strategy,
+			TargetFile: graph.File,
+			Deps:       ApiFormatDeps(graph.Graph),
+		})
+	}
+
+	return result
+}
+
+func ApiFormatDeps(deps graph.Deps) []ApiDep {
```

Comment:
I'm not sure I agree with modifying this part of the upload. In order to move away from the current upload format, I think we need to have a very strong reason and I'm not sure I see one. I'm hesitant to go down this path due to the difficulty that the backend would have in creating the graph